### PR TITLE
Add Compound Mapping

### PR DESCRIPTION
**Pull request by mubaris**
Comparing to `deployment` branch

### CODE REVIEW

File: contracts/connectors/compound/main.sol [hunk 1]
```diff
@@ -47,12 +49,13 @@ abstract contract CompoundResolver is Events, Helpers {
     */
     function withdraw(
         address token,
+        string calldata tokenId,
```
Comment (thrilok209) [inline contracts/connectors/compound/main.sol:52]: @mubaris We are going to pass only tokenId. And from tokenId we can get ctoken and token address

File: contracts/mapping/compound.sol [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        require(_names.length == _ctokens.length, "updateCtokenMapping: not same length");
+
+        for (uint i = 0; i < _ctokens.length; i++) {
+            require(tokenMapping[_names[i]] != address(0), "updateCtokenMapping: mapping does not exist");
+            require(cTokenMapping[_names[i]] != address(0), "updateCtokenMapping: mapping does not exist");
+
+            require(_tokens[i] != address(0), "updateCtokenMapping: _tokens address not vaild");
+            require(_ctokens[i] != address(0), "updateCtokenMapping: _ctokens address not vaild");
+
+            require(CTokenInterface(_ctokens[i]).isCToken(), "updateCtokenMapping: not a cToken");
```
Comment (thrilok209) [inline contracts/mapping/compound.sol:78]: @mubaris Do we have this function in all the ctokens?

File: contracts/mapping/compound.sol [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    event LogCTokensAdded(string[] names, address[] tokens, address[] ctokens);
+    event LogCTokensUpdated(string[] names, address[] tokens, address[] ctokens);
+
+    ConnectorsInterface public immutable connectors;
+
+    // InstaIndex Address.
+    IndexInterface public constant instaIndex = IndexInterface(0x2971AdFa57b20E5a416aE5a708A8655A9c74f723);
+
+    mapping (string => address) public cTokenMapping;
+    mapping (string => address) public tokenMapping;
```
Comment (thrilok209) [inline contracts/mapping/compound.sol:27]:
```suggestion
   struct TokenMap {
       address ctoken;
       address token;
   }

    mapping (string => TokenMap) public cTokensMapping;
```

File: contracts/mapping/compound.sol (hunk 2 above)
Comment (mubaris) [reply contracts/mapping/compound.sol:78]: All the existing ones have it, and it's part of the cToken interface. So I assume the new ones will have also this

https://github.com/compound-finance/compound-protocol/blob/b198cb4dac977c61fa793ffe441c932438e83cdc/contracts/CTokenInterfaces.sol#L124

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**